app/models/featured_topic.rb

```diff
@@ -279,6 +279,14 @@ def slug
     category_key.to_s.dasherize
   end
 
+  def path
+    if definition.has_key?(:path)
+      definition[:path]
+    else
+      Rails.application.routes.url_helpers.featured_topic_path(slug)
```

Comment:
This is a confusing topic, but i feel like I read somewhere that there are performance problems with calling `Rails.application.routes.url_helpers`, even though everyone did it. There is some alternate way to do it, that is like doing an `include` into a class or something....

Alternately, it might not be so bad just to hard-code `/focus/#{slug}` or whatever. Alternately, it's not so bad to have the logic in the view `if path`. 

Wanting view-related logic in a model... is why we have those "ViewModels", hopefully eventually to be replaced by ViewComponent. Once we are using ViewComponent I think since it works more smoothly we'll end up using it in even more places, but for now not necessarily suggesting adding that complexity, just pointing out that it's motivation is related to all this.

Reply:
OK. I had my doubts about it. I'm going with just hard-coding it, along with a comment.

Reply:
and actually, you can write a test that tests that it *matches*  `featured_topic_path`, so if the assumption ever changes, the test will warn us!